generated/mocks/generate.go

```diff
@@ -0,0 +1,24 @@
+// Copyright (c) 2018 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+// mockgen rules for generating mocks using file mode.
+//go:generate sh -c "mockgen -package=store -destination=$GOPATH/src/$PACKAGE/service/r2/store/storage_mock.go -source=$GOPATH/src/$PACKAGE/service/r2/store/store.go"
```

Comment:
Perhaps just `$GOPATH/src/$PACKAGE/service/r2/store/mock.go` instead of `$GOPATH/src/$PACKAGE/service/r2/store/storage_mock.go` since it's under the `store` package so the name is self explanatory already?

Reply:
Also I think you should be able to use reflection mode? If so, you can pipe the output to `mockclean` to fix the import ordering, e.g., `https://github.com/m3db/m3aggregator/blob/master/generated/mocks/generate.go#L22`.

Reply:
Ping